README.md

```diff
@@ -1,35 +1,116 @@
-## How it works
+# Connect
 
-The Connectivity component is responsible for the communication with the target system. This component implements the logic of the interaction protocol, receives and sends messages from and to the system, respectively.
+The "Connect" component is responsible for the communication with a target system.
+This component implements the logic of the interaction protocol, receiving and sending messages from and to the system, respectively.
 
-## Queues
+This project includes only an adapter for using the Sailfish service in the th2 packed into a Docker Image.
+This image should be used as a base to implement extensions with the real logic for specific protocols using services in the Sailfish format.
 
-Connectivity has 4 queues for interacting with th2 components. Messages that were received from / sent to the target system will be sent to the following queues:
+As an example, the [th2-conn-generic](https://github.com/th2-net/th2-conn-generic) project implements the extension for connecting via FIX protocol using standard Sailfish's FIX service.
 
-- incoming parsed messages
-- outgoing parsed messages
+## Configuration
+
+This configuration should be specified in the custom configuration block in schema editor.
+
+```yaml
+session-alias: "connectivity-alias"
+workspace: "/home/sailfish/workspace"
+type: "th2_service:Your_Service_Type"
+name: "your_service"
+settings:
+  param1: "value1"
+```
+
+Parameters:
++ session-alias - that session alias will be set for all messages received or sent by this component. **It should be unique for each "Connect" component**;
++ workspace - the folder inside the container that will contain a plugin adapted to use in the TH2;
++ type - the service type from **services.xml** file. If service name from services.xml file contains `-` symbols they must be replaced with `_` symbol;
++ name - the service name that will be displayed in the events in the report;
++ settings - the parameters that will be transformed to the actual service's settings specified in the **services.xml** file.
+
+## Extension
+
+You can add the ability to connect to a target system by implementing your own service in the Sailfish format and putting it and its configuration to the correct places in the base image.
+
+You need to take the following steps:
+
+1. Create the implementation of the [com.exactpro.sf.services.IService](https://github.com/exactpro/sailfish-core/blob/master/BackEnd/Core/sailfish-core/src/main/java/com/exactpro/sf/services/IService.java).
+The examples of implementing this interface can be found [here](https://github.com/exactpro/sailfish-core/tree/master/BackEnd/Service).
+If the protocol is already implemented in the Sailfish services you can simply use the dependencies on the service's artifact for that protocol.
+
+2. Create the **services.xml** configuration file that contains the description for services you can use from that "Connect" component.
```

Comment:
Create the services.xml configuration file **which** contains the description for **the** services **that** you can use from **the** "Connect" component.